### PR TITLE
[.NET] Adopt c# 10 file scoped namespaces

### PR DESCRIPTION
**Pull request by Romfos**
### 🤔 What's changed?

Adopt c# 10 file scoped namespaces

documentation: https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-10.0/file-scoped-namespaces

### ⚡️ What's your motivation? 

Adopt modern c# language feature to simplify code base

### 🏷️ What kind of change is this?

- :bank: Refactoring/debt/DX (improvement to code design, tooling, etc. without changing behaviour)

### ♻️ Anything particular you want feedback on?

no changes in product. code style changes only

### 📋 Checklist:

- [x] I agree to respect and uphold the [Cucumber Community Code of Conduct](https://github.com/cucumber/.github/tree/main?tab=coc-ov-file)
- [ ] I've changed the behaviour of the code
  - [ ] I have added/updated tests to cover my changes.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly.
- [ ] Users should know about my change
  - [x] I have added an entry to the "Unreleased" section of the [**CHANGELOG**](../blob/main/CHANGELOG.md), linking to this pull request.

----

*This text was originally generated from a [template](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/about-issue-and-pull-request-templates), then edited by hand. [You can modify the template here.](https://github.com/cucumber/.github/edit/main/.github/PULL_REQUEST_TEMPLATE.md)*

### CODE REVIEW

**Comment by Romfos** [comment]
Parser.cs is not changed because it has header
```
// ------------------------------------------------------------------------------
//  <auto-generated>
//      This code was generated by Berp (http://https://github.com/gasparnagy/berp/).
// 
//      Changes to this file may cause incorrect behavior and will be lost if
//      the code is regenerated.
//  </auto-generated>
// ------------------------------------------------------------------------------
```

**Comment by mpkorstanje** [comment]
You can edit [dotnet/gherkin-csharp.razor](https://github.com/cucumber/gherkin/blob/main/dotnet/gherkin-csharp.razor) to apply the name spaces to the generated code. Instruction on how to generate the parser can be found in[ CONTRIBUTING.md](https://github.com/cucumber/gherkin/blob/main/CONTRIBUTING.md#generating-parsers).

**Review by clrudolphi — COMMENTED** [review]
Other than the 1 spacing/formatting item found (commented elsewhere), this looks good.
It would be nice, for consistency's sake, to complete this PR by adding in the changes required of the razor file that @mpkorstanje suggested. I can walk you through them if you need it.

File: dotnet/Gherkin/TokenMatcher.cs [hunk 1]
```diff
@@ -3,231 +3,230 @@
...
 using Gherkin.Ast;
 
-namespace Gherkin
+namespace Gherkin;
+
+public class TokenMatcher : ITokenMatcher
 {
-    public class TokenMatcher : ITokenMatcher
-    {
 		private readonly Regex LANGUAGE_PATTERN = new Regex ("^\\s*#\\s*language\\s*:\\s*([a-zA-Z\\-_]+)\\s*$");
```
[the remark this answers is not in the document]
Comment (Romfos) [reply dotnet/Gherkin/TokenMatcher.cs]: I did run dotnet format to fix spacing in whole solution

**Comment by Romfos** [comment]
please squash this PR during the merge

**Comment by Romfos** [comment]
I propose to make changes in code-generation later if needed, because I'm planning to propose another changes in this library, including code style

**Pull request merged**